packages/db/src/search/StatsSearch.ts

```diff
@@ -0,0 +1,298 @@
+    );
+    const aggregateQueryRes: TotalSummary[] = await aggregateQuery;
+    return this.formatSummaryResult(aggregateQueryRes[0]);
+  }
+
+  /** Performs joins on the provider and/or collection table if neccessary
+   *
+   * @param {Knex} knex - the knex client to be used
+   * @returns {Knex.QueryBuilder} the knex query of a joined table or not based on queryStringParams
+   */
+  private providerAndCollectionIdBuilder(knex: Knex): Knex.QueryBuilder {
+    let aggregateQuery;
+    this.queryStringParameters.field = this.queryStringParameters.field ? this.queryStringParameters.field : 'status';
+    if (this.queryStringParameters.field?.includes('error.Error')) {
+      aggregateQuery = knex.select(knex.raw(`"count(${typeToTable[this.type]}.cumulus_id), error ->> 'Error' as error"`)).from(typeToTable[this.type]);
+    } else {
+      aggregateQuery = knex.select(`${typeToTable[this.type]}.${this.queryStringParameters.field}`).from(typeToTable[this.type]);
+    }
+    if (this.queryStringParameters.collectionId) {
+      aggregateQuery.join(`${TableNames.collections}`, `${typeToTable[this.type]}.collection_cumulus_id`, 'collections.cumulus_id');
+    }
+
+    if (this.queryStringParameters.provider) {
+      aggregateQuery.join(`${TableNames.providers}`, `${typeToTable[this.type]}.provider_cumulus_id`, 'providers.cumulus_id');
+    }
+    return aggregateQuery;
+  }
+
+  /** Aggregates the search query based on queryStringParameters
+   *
+   * @param {Knex.QueryBuilder} query - the knex query to be aggregated
+   * @param {Knex} knex - the knex client to be used
+   * @returns {Knex.QueryBuilder} the query with its new Aggregatation
+   */
+  private aggregateQueryField(query: Knex.QueryBuilder, knex: Knex): Knex.QueryBuilder {
+    this.queryStringParameters.field = this.queryStringParameters.field ? this.queryStringParameters.field : 'status';
+    if (this.queryStringParameters.field?.includes('error.Error')) {
+      query.select(knex.raw("error ->> 'Error' as error"))
+        .count(`${typeToTable[this.type]}.cumulus_id as count`)
+        .groupByRaw(knex.raw("error ->> 'Error'"))
+        .orderBy('count', 'desc');
+    } else {
+      query.select(`${typeToTable[this.type]}.${this.queryStringParameters.field}`)
+        .count(`${typeToTable[this.type]}.cumulus_id as count`)
+        .groupBy(`${typeToTable[this.type]}.${this.queryStringParameters.field}`)
+        .orderBy('count', 'desc');
+    }
+    return query;
+  }
+
+  /**
+   * Builds basic query
+   *
+   * @param knex - the knex client
+   * @returns the search query
+   */
+  protected buildBasicQuery(knex: Knex)
+    : {
+      searchQuery: Knex.QueryBuilder,
+    } {
+    let searchQuery:Knex.QueryBuilder;
+    if (this.queryStringParameters.provider || this.queryStringParameters.collectionId) {
+      searchQuery = this.providerAndCollectionIdBuilder(knex);
+    } else {
+      searchQuery = knex(`${typeToTable[this.type]}`);
+    }
```

Comment:
I think this would be cleaner:
Const searchQuery:Knex.QueryBuilder = knex(`${typeToTable[this.type]}`);
this.providerAndCollectionIdBuilder(searchQuery);